sematic/future_context.py

```diff
@@ -0,0 +1,88 @@
+# Standard Library
+from contextlib import contextmanager
+from dataclasses import dataclass
+from importlib import import_module
+from typing import Any, Optional, Type
+
+
+@dataclass(frozen=True)
+class SematicContext:
+    """Contextual information about the execution of the current Sematic function
+    Attributes
+    ----------
+    id:
+        The id of the future for the current execution. For cloud executions, this
+        is equivalent to the id for the existing run.
+    root_id:
+        The id of the root future for a resolution. For cloud executions, this is
+        equivalent to the id for the root run.
+    resolver_class_path:
+        The import path for the resolver being used.
+    """
+
+    id: str
+    root_id: str
```

Comment:
Should these be named `run_id` and `resolution_id`?

I'm less adamant about `root_id`, but at least `id` should be renamed. As a user I would say "what is this the ID of?". Granted this is user-facing, which I believe it is?

Reply:
side note: `id -> run_id` does mean the first introduction of something referencing the concept of "run" within the silent resolver. But I'm willing to compromise on that!